src/UserDocument/UserDocumentManager.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace AppBundle\UserDocument;
+
+use AppBundle\Entity\UserDocument;
+use Doctrine\Common\Persistence\ObjectManager;
+use League\Flysystem\Filesystem;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class UserDocumentManager
+{
+    private $validator;
+    private $manager;
+    private $storage;
+
+    public function __construct(ValidatorInterface $validator, ObjectManager $manager, Filesystem $storage)
+    {
+        $this->validator = $validator;
+        $this->manager = $manager;
+        $this->storage = $storage;
+    }
+
+    public function createAndSave(UploadedFile $uploadedFile, string $type): UserDocument
+    {
+        $document = UserDocument::createFromUploadedFile($uploadedFile);
+        $document->setType($type);
+
+        $errors = $this->validator->validate($document);
+
+        if (count($errors) > 0) {
+            $errMsg = '';
+            foreach ($errors as $error) {
+                $errMsg .= $error->getMessage().' ';
+            }
+
+            throw new \Exception($errMsg);
```

Comment:
Throw a more specific exception please otherwise we are not able to catch this very specific behavior only.